llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -26219,6 +26223,21 @@ SDValue DAGCombiner::visitFP_TO_BF16(SDNode *N) {
   return SDValue();
 }
 
+SDValue DAGCombiner::visitBF16_TO_FP(SDNode *N) {
```

Comment:
Combine this with visitFP16_TO_FP above.

But couldn't SimplifyDemandedBits handle this optimization in a more generic way?

Reply:
You need to use `APInt DemandedBits = APInt::getLoBitsSet(N0.getScalarValueSizeInBits(), 16);` so that only the lower 16 bits are demanded.